secant/Util/Clamped.swift

```diff
@@ -16,16 +16,17 @@ struct Clamped<Value: Comparable> {
     let range: ClosedRange<Value>
     var wrappedValue: Value {
         get { value }
-        set {
-            value = min(
-                max(range.lowerBound, newValue),
-                range.upperBound
-            )
-        }
+        set { value = clamp(value: newValue, range: range) }
     }
 
     init(wrappedValue: Value, _ range: ClosedRange<Value>) {
         self.value = wrappedValue
         self.range = range
+        
+        value = clamp(value: wrappedValue, range: range)
```

Comment:
Is this duplicating the setter already happening on line 23?

Reply:
Yeah, this is unfortunate but the clamp function can't be used before being initialised. So both value and range need to be set first in order to call clamp, hence "double" set. Aka set the value to the input one and clamp it later once initialised.